Failing tests in test_bioformats.py

Tests are failing in `test_bioformats.py`. I noticed this while working on PR https://github.com/soft-matter/pims/pull/403, but it seems to happen for all branches, so I don't think it's caused by my code.

To create a test environment, including dependencies for video files:
```
conda create -n pims-dev python=3.9 pip ipython
conda activate pims-dev
cd GitHub/pims
python -m pip install -r requirements-dev.txt
python -m pip install -e .
conda install -c conda-forge av imageio-ffmpeg moviepy tifffile
```

To run the tests:
```
python download_bioformats_test.py
pytest pims/tests/test_bioformats.py -vv
```

Problems encountered:
* Inconsistent failing tests. In `test_bioformats.py` I see a lot of `test_repr` failures, but when I run just the single test it passes fine. Commenting out `test_repr` leads to other tests failing (that again, pass fine when run individually). Is something weird happening where the test files are not being closed properly? I don't know how to debug this.
* There's something wrong with the norpix reader. It can't open the `HEART.SEQ` test image file. Oddly, all of the tests in `pytest pims/tests/test_norpix.py` pass. But `pytest pims/tests/test_bioformats::TestBioformatsSEQ::test_open` fails.

More weird stuff I have noticed:
* Only the PyAV reader can open the file `pims/tests/data/bioformats/wtembryo.mov`. If PyAV is not available, sometimes other handlers try and fail (which casues a test failure). Given that this file will play in VLC but not QuickTime, I assume that's something to do with the codecs? I don't think this is a problem, because at least *something* can open the file, but it does mean you must have PyAV installed as a dependency for the tests not to fail.

Some good news, we're down to one failing test as far as I can tell.

Result of `pytest pims/tests/test_bioformats.py -vv`
`TestBioformatsSEQ.test_open` is the one currently failing.

<details>
<summary>pytest report:</summary>

```
=================================== FAILURES ===================================
_________________________ TestBioformatsSEQ.test_open __________________________

cls = <class '_pytest.runner.CallInfo'>
func = <function call_runtest_hook.<locals>.<lambda> at 0x13e3bede0>
when = 'call'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: "Callable[[], TResult]",
        when: "Literal['collect', 'setup', 'call', 'teardown']",
        reraise: Optional[
            Union[Type[BaseException], Tuple[Type[BaseException], ...]]
        ] = None,
    ) -> "CallInfo[TResult]":
        """Call func, wrapping the result in a CallInfo.

        :param func:
            The function to call. Called without arguments.
        :param when:
            The phase in which the function is called.
        :param reraise:
            Exception or exceptions that shall propagate if raised by the
            function, instead of being wrapped in the CallInfo.
        """
        excinfo = None
        start = timing.time()
        precise_start = timing.perf_counter()
        try:
>           result: Optional[TResult] = func()

../../../mambaforge/envs/pims/lib/python3.11/site-packages/_pytest/runner.py:339:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../mambaforge/envs/pims/lib/python3.11/site-packages/_pytest/runner.py:260: in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
../../../mambaforge/envs/pims/lib/python3.11/site-packages/pluggy/_hooks.py:265: in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
../../../mambaforge/envs/pims/lib/python3.11/site-packages/pluggy/_manager.py:80: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
../../../mambaforge/envs/pims/lib/python3.11/site-packages/_pytest/unraisableexception.py:88: in pytest_runtest_call
    yield from unraisable_exception_runtest_hook()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def unraisable_exception_runtest_hook() -> Generator[None, None, None]:
        with catch_unraisable_exception() as cm:
            yield
            if cm.unraisable:
                if cm.unraisable.err_msg is not None:
                    err_msg = cm.unraisable.err_msg
                else:
                    err_msg = "Exception ignored in"
                msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
                msg += "".join(
                    traceback.format_exception(
                        cm.unraisable.exc_type,
                        cm.unraisable.exc_value,
                        cm.unraisable.exc_traceback,
                    )
                )
>               warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
E
E               Traceback (most recent call last):
E                 File "/Users/genevieb/Documents/GitHub/pims/pims/api.py", line 203, in open
E                   messages.append('{} errored: {}'.format(str(handler), str(e)))
E               ResourceWarning: unclosed file <_io.BufferedReader name='/Users/genevieb/Documents/GitHub/pims/pims/tests/data/bioformats/HEART.SEQ'>

../../../mambaforge/envs/pims/lib/python3.11/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning
=========================== short test summary info ============================
FAILED pims/tests/test_bioformats.py::TestBioformatsSEQ::test_open - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO ...
======================== 1 failed, 169 passed in 35.74s ========================
```

</details>

--- a/pims/cine.py
+++ b/pims/cine.py
@@ -336,7 +336,6 @@ class Cine(FramesSequence):
     For a .chd file, this class only reads the header, not the images.
     
     """
-    # TODO: Unit tests using a small sample cine file.
     @classmethod
     def class_exts(cls):
         return {'cine'} | super(Cine, cls).class_exts()
@@ -742,6 +741,10 @@ class Cine(FramesSequence):
 
     def close(self):
         self.f.close()
+
+    def __del__(self):
+        if hasattr(self, 'f'):
+            self.f.close()
 
     def __unicode__(self):
         return self.filename

--- a/pims/imageio_reader.py
+++ b/pims/imageio_reader.py
@@ -149,3 +149,7 @@ class ImageIOReader(FramesSequenceND):
     def close(self):
         self.reader.close()
         super().close()
+
+    def __del__(self):
+        if hasattr(self, 'reader'):
+            self.reader.close()

--- a/pims/norpix_reader.py
+++ b/pims/norpix_reader.py
@@ -256,6 +256,10 @@ class NorpixSeq(FramesSequence):
     def close(self):
         self._file.close()
 
+    def __del__(self):
+        if hasattr(self, '_file'):
+            self._file.close()
+
     def __repr__(self):
         return """<Frames>
 Source: {filename}

